homeassistant/components/alexa/handlers.py

```diff
@@ -232,7 +232,11 @@ async def async_api_turn_off(
 
     service = SERVICE_TURN_OFF
     if entity.domain == cover.DOMAIN:
-        service = cover.SERVICE_CLOSE_COVER
+        supported = entity.attributes.get(ATTR_SUPPORTED_FEATURES, 0)
+        if supported & cover.CoverEntityFeature.STOP:
+            service = cover.SERVICE_STOP_COVER
```

Comment:
May be we also should check if the cover's state is `closing` to allow the cover to stop.

Reply:
I would like to not include that check for 2 reasons:
- It seems kind of counter intuitive to me that the "turn off" command would stop the cover or close it, depending on its state (I left the old implementation of closing for backwards compatibility, but for a cover that supports stopping I think "turn off" is most reasonably mapped to stop)
- Covers don't always report their state, in fact the cover I'm testing this with doesn't because it's an MQTT cover (talking to an RF bridge) which is unable to report its state back

Do you think there could be any ill effects in allowing covers to stop when they are not moving?

Reply:
I understand what you want, but we must be very careful changing the behavior. May be you are right. Still my question stands to share the state of your cover entity where it is opening or closing and you want to stop it.
At least changing this behavior makes this change a breaking change.
But at some point it makes sense as we are talking about the power controller here.

Reply:
The state is either "opening" or "closing" (when I initiated the movement via homeassistant) or whatever it was before ("open", "closed") depending on what it previously was, when the movement was not initiated via homeassistant. To further complicate this, the RF commands sometimes don't reach the cover, which leads to a state change in homeassistant but no real movement. Thus the state is totally unreliable and can't be trusted

Reply:
May we we could only support it if `CoverEntityFeature.CLOSE` and `CoverEntityFeature.OPEN` or `CoverEntityFeature.SET_POSITION`. Just to  make sure it is still possible to open and close the cover.
In that case the will have a mode controller or a range controller, allowing to open and close the cover / place it a certain position. So in that case the power controller `off` triggered by `stop` could be handled as a stop.

If you agree, could you add such a check and the tests needed?

Also add re-add the breaking change section.

Reply:
Hmm interesting, yeah, I think checking whether (`OPEN` and `CLOSE`) or (`SET_POSITION`) are supported is a good idea.

I'll try adding appropriate tests